common/internal_model/src/test/scala/uk/ac/wellcome/models/work/generators/WorkGenerators.scala

```diff
@@ -0,0 +1,136 @@
+package uk.ac.wellcome.models.work.generators
+
+import uk.ac.wellcome.models.work.internal._
+import WorkState._
+
+trait WorkGenerators extends IdentifiersGenerators {
+
+  def sourceWork(
```

Comment:
What determines which are intrinsic properties (ie the parameters here) vs ones to be set via the fluent interface?

Reply:
Anything in `state` can't be in the fluent interface. I could move the `version` outside of the parameters to the fluent interface too if you think that would make things better?

Reply:
:+1: will do that in any case so that the intitial function creates a work with some state and the subsequent operations apply to the work independent of its state.